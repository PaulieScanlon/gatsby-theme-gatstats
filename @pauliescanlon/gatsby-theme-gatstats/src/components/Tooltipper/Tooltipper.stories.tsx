import * as React from 'react'

import { Tooltipper } from './Tooltipper'

export default {
  title: 'Tooltipper',
  parameters: {
    component: Tooltipper,
    componentSubtitle: 'Tooltipper is...'
  }
}

export const usage = () => (
  <Tooltipper title="Jan" value={10} left={20} top={60} />
)
