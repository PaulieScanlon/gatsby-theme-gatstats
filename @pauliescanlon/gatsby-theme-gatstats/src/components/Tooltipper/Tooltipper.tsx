/** @jsx jsx */
import { jsx } from 'theme-ui'

import { Tooltip } from '@vx/tooltip'

interface ITooltipProps {
  /** The tooltip title */
  title: string
  /** The tooltip value */
  value: number | string
  /** Left position required by @vx/toottip */
  left: number
  /** Top position required by @vx/toottip */
  top: number
}

export const Tooltipper: React.FC<ITooltipProps> = ({
  title,
  value,
  left,
  top
}) => {
  return (
    <Tooltip
      left={`${left - 3}px`}
      top={`${top - 15}px`}
      style={{
        backgroundColor: 'none',
        padding: 0,
        margin: 0,
        boxShadow: 'none'
      }}
    >
      <span
        sx={{
          position: 'relative',
          color: 'primary',
          backgroundColor: 'background',
          display: 'inline-flex',
          fontFamily: 'body',
          fontSize: 0,
          p: 2,
          borderRadius: 0,
          borderWidth: 0,
          borderStyle: 'solid',
          borderColor: 'primary',
          ':after': {
            content: `''`,
            position: 'absolute',
            width: '12px',
            height: '12px',
            top: '100%',
            left: '50%',
            transform: 'translate(-50%,-50%) rotate(45deg)',
            backgroundColor: 'background',
            borderBottomWidth: 0,
            borderBottomStyle: 'solid',
            borderBottomColor: 'primary',
            borderRightWidth: 0,
            borderRightStyle: 'solid',
            borderRightColor: 'primary'
          },
          span: {
            mr: 1
          }
        }}
      >
        <span>{`${title}:`}</span>
        <strong>{`x${value}`}</strong>
      </span>
    </Tooltip>
  )
}
