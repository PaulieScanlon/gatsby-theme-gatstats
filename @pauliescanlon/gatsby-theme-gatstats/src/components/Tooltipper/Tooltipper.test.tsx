import * as React from 'react'
import { shallow } from 'enzyme'

import { ThemeProvider } from 'theme-ui'
import theme from '../../gatsby-plugin-theme-ui'

import { usage } from './Tooltipper.stories'

describe('<Tooltipper>', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <ThemeProvider theme={theme}>{usage()}</ThemeProvider>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
