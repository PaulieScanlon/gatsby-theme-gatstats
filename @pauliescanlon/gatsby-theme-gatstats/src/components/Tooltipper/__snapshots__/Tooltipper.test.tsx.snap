// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tooltipper> renders correctly 1`] = `
<RootProvider
  theme={
    Object {
      "borderWidths": Array [
        1,
        4,
      ],
      "breakpoints": Array [
        "576px",
        "768px",
        "992px",
        "1200px",
      ],
      "colors": Object {
        "background": "#131127",
        "modes": Object {
          "light": Object {
            "background": "#ffffff",
            "primary": "#f98ddc",
            "secondary": "#87bff5",
            "surface": "#fafafa",
            "text": "#111111",
            "textMuted": "#d2d5f2",
            "textSecondary": "#333333",
            "transparent": "rgba(0,0,0,0)",
          },
        },
        "primary": "#a92aeb",
        "secondary": "#688ce0",
        "surface": "#1a1832",
        "text": "#ffffff",
        "textMuted": "#4e4c7a",
        "textSecondary": "#d2d2f1",
        "transparent": "rgba(0,0,0,0)",
      },
      "fontSizes": Array [
        12,
        14,
        16,
        18,
        20,
        24,
        32,
        48,
        64,
      ],
      "fontWeights": Object {
        "body": 400,
        "bold": 700,
        "heading": 700,
      },
      "fonts": Object {
        "body": "system-ui, sans-serif",
        "heading": "system-ui, sans-serif",
        "monospace": "Menlo, monospace",
      },
      "initialColorModeName": "dark",
      "radii": Array [
        4,
        8,
        50,
      ],
      "shadows": Array [
        "0 0 0 2px",
        "0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.20)",
        "0 3px 6px rgba(0,0,0,0.16), 0 3px 6px rgba(0,0,0,0.19)",
        "0 10px 20px rgba(0,0,0,0.19), 0 6px 6px rgba(0,0,0,0.19)",
        "0 14px 28px rgba(0,0,0,0.25), 0 10px 10px rgba(0,0,0,0.18)",
        "0 19px 38px rgba(0,0,0,0.30), 0 15px 12px rgba(0,0,0,0.17)",
      ],
      "space": Array [
        0,
        4,
        8,
        16,
        24,
        32,
        48,
        64,
      ],
      "styles": Object {
        "a": Object {
          ":focus": Object {
            "color": "text",
          },
          ":hover": Object {
            "color": "text",
          },
          "color": "primary",
          "cursor": "pointer",
          "fontFamily": "body",
          "fontSize": Array [
            2,
            3,
          ],
          "fontWeight": "body",
          "lineHeight": "28px",
          "outline": "none",
          "transition": ".2s linear all",
        },
        "blockquote": Object {
          "backgroundColor": "background",
          "borderColor": "secondary",
          "borderLeftStyle": "solid",
          "borderLeftWidth": 4,
          "mx": Array [
            2,
            3,
          ],
          "my": 3,
          "p": Object {
            "mb": 0,
          },
          "px": 3,
          "py": 3,
        },
        "h1": Object {
          "a": Object {
            "color": "inherit",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "lineHeight": "inherit",
          },
          "color": "text",
          "fontFamily": "heading",
          "fontSize": Array [
            6,
            7,
            8,
          ],
          "fontWeight": "heading",
          "lineHeight": "normal",
          "margin": 0,
          "marginBottom": 3,
          "wordBreak": "break-word",
        },
        "h2": Object {
          "a": Object {
            "color": "inherit",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "lineHeight": "inherit",
          },
          "color": "text",
          "fontFamily": "heading",
          "fontSize": Array [
            5,
            7,
          ],
          "fontWeight": "heading",
          "lineHeight": "normal",
          "margin": 0,
          "marginBottom": 3,
          "wordBreak": "break-word",
        },
        "h3": Object {
          "a": Object {
            "color": "inherit",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "lineHeight": "inherit",
          },
          "color": "text",
          "fontFamily": "heading",
          "fontSize": Array [
            4,
            6,
          ],
          "fontWeight": "heading",
          "lineHeight": "normal",
          "margin": 0,
          "marginBottom": 3,
          "wordBreak": "break-word",
        },
        "h4": Object {
          "a": Object {
            "color": "inherit",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "lineHeight": "inherit",
          },
          "color": "text",
          "fontFamily": "heading",
          "fontSize": Array [
            4,
            5,
          ],
          "fontWeight": "heading",
          "lineHeight": "normal",
          "margin": 0,
          "marginBottom": 3,
          "wordBreak": "break-word",
        },
        "h5": Object {
          "a": Object {
            "color": "inherit",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "lineHeight": "inherit",
          },
          "color": "text",
          "fontFamily": "heading",
          "fontSize": Array [
            3,
            4,
          ],
          "fontWeight": "heading",
          "lineHeight": "normal",
          "margin": 0,
          "marginBottom": 2,
          "wordBreak": "break-word",
        },
        "h6": Object {
          "a": Object {
            "color": "inherit",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "lineHeight": "inherit",
          },
          "color": "text",
          "fontFamily": "heading",
          "fontSize": Array [
            2,
            3,
          ],
          "fontWeight": "body",
          "lineHeight": "normal",
          "margin": 0,
          "marginBottom": 2,
          "wordBreak": "break-word",
        },
        "hr": Object {
          "opacity": 0,
        },
        "img": Object {
          "height": "auto",
          "width": "100%",
        },
        "li": Object {
          "color": "text",
          "fontFamily": "body",
          "fontSize": Array [
            1,
            2,
          ],
          "fontWeight": "body",
          "lineHeight": "28px",
          "p": Object {
            "color": "text",
          },
        },
        "ol": Object {
          "marginTop": 3,
          "paddingLeft": 3,
        },
        "p": Object {
          "a": Object {
            ":focus": Object {
              "color": "text",
            },
            ":hover": Object {
              "color": "text",
            },
            "color": "primary",
            "cursor": "pointer",
            "fontFamily": "body",
            "fontSize": Array [
              2,
              3,
            ],
            "fontWeight": "body",
            "lineHeight": "28px",
            "outline": "none",
            "transition": ".2s linear all",
          },
          "code": Object {
            "WebkitFontSmoothing": "antialiased",
            "backgroundColor": "surface",
            "borderRadius": 0,
            "fontFamily": "monospace",
            "fontSize": Array [
              2,
              3,
            ],
            "fontWeight": "body",
            "lineHeight": "28px",
            "p": 1,
          },
          "color": "textSecondary",
          "fontFamily": "body",
          "fontSize": Array [
            2,
            3,
          ],
          "fontWeight": "body",
          "lineHeight": "28px",
          "marginBottom": 4,
          "marginTop": 0,
          "wordBreak": "break-word",
        },
        "root": Object {
          "fontFamily": "body",
          "fontWeight": "body",
          "lineHeight": "normal",
        },
        "table": Object {
          "borderCollapse": "separate",
          "borderSpacing": 0,
          "color": "textSecondary",
          "fontFamily": "body",
          "width": "100%",
        },
        "td": Object {
          "borderColor": "textMuted",
          "borderStyle": "solid",
          "borderWidth": 0,
          "px": 2,
          "py": 2,
        },
        "th": Object {
          "backgroundColor": "background",
          "borderBottomColor": "transparent",
          "borderLeftColor": "textMuted",
          "borderRightColor": "textMuted",
          "borderStyle": "solid",
          "borderTopColor": "textMuted",
          "borderWidth": 0,
          "color": "text",
          "px": 2,
          "py": 2,
        },
        "tr": Object {},
        "ul": Object {
          "marginTop": 3,
          "paddingLeft": 3,
        },
      },
      "transitions": Object {
        "sideBarTransition": ".3s ease-in-out margin-left",
      },
    }
  }
>
  <Tooltipper
    left={20}
    title="Jan"
    top={60}
    value={10}
  />
</RootProvider>
`;
